server/motd.lua

```diff
@@ -0,0 +1,96 @@
+local acknowledged = GetConvar('qbx:acknowledge', 'false') == 'true'
+local messagesUrl = GetConvar('qbx:serviceMessagesUrl', 'https://raw.githubusercontent.com/Qbox-project/txAdminRecipe/refs/heads/main/service-messages.json')
+local resourceVersion = GetResourceMetadata(cache.resource, 'version', 0)
+
+---@param str string
+---@param prefix string
+---@return boolean
+local function startsWith(str, prefix)
+    return str:sub(1, prefix:len()) == prefix
+end
+
+---Returns whether the given version prefix or exact version matches the resource's version.
+---@return boolean
+local function isResourceVersion(version)
```

Comment:
May want to either get rid of the recursion or add a clear param type for version so that it's clear that a table of tables wouldn't be expected.

Reply:
Technically it *can* work. The problem with adding a clear param type is that we're working with JSON objects, in which case the real type isn't really known. It's also why I handled multiple different types for `version` and `content`.